contracts/AuraLiquidityMigrator.sol

```diff
@@ -0,0 +1,194 @@
+ *          then it creates or join to a balancer pool, adds liquidity to the pool.
+ *          Finally if the pool already existed it stakes the liquidity in an aura reward pool.
+ */
+contract AuraLiquidityMigrator {
+    using SafeERC20 for IERC20;
+    /// @dev Balancer pool factory
+    IWeightedPoolFactory public immutable BALANCER_POOL_FACTORY; // ie 0x8E9aa87E45e92bad84D5F8DD1bff34Fb92637dE9
+    /// @dev Balancer vault
+    IVault public immutable BALANCER_VAULT; // ie 0xBA12222222228d8Ba445958a75a0704d566BF2C8
+
+    /**
+     * @param _weightedPoolFactory The pool factory address
+     * @param _balancerVault The balancer vault address
+     */
+    constructor(address _weightedPoolFactory, address _balancerVault) {
+        BALANCER_POOL_FACTORY = IWeightedPoolFactory(_weightedPoolFactory);
+        BALANCER_VAULT = IVault(_balancerVault);
+    }
+
+    event PoolCreated(address indexed pool);
+
+    /**
+     * @dev Migrates a liquidity position, deploys a new `WeightedPool` and adds liquidity to the pool.
+     *      The created pool has a weight of 50/50.
+     *
+     * @param fromLpToken The LP token to migrate
+     * @param user The owner of the lp position
+     * @param name The name of the new pool
+     * @param symbol The symbol of the new pool
+     * @param tokens The underlying tokens of the pool.
+     * @param swapFeePercentage The swap fee percentage
+     * @param owner The owner of the pool
+     * @param minOut The min amount of bpt
+     */
+    function migrateUniswapV2AndCreatePool(
+        address fromLpToken, //Uniswap LP token
+        address user,
+        string memory name,
+        string memory symbol,
+        IERC20[] memory tokens,
+        uint256 swapFeePercentage,
+        address owner,
+        uint256 minOut
+    ) external returns (address pool) {
+        require(tokens.length == 2, "only token pairs");
+        // 1. Remove liquidity
+        (uint256 amount0, uint256 amount1) = _removeLiquidityUniswapV2(fromLpToken, user);
+
+        // 2. Create balancer pool
+        pool = _createBalancerPool(name, symbol, tokens, swapFeePercentage, owner);
+
+        // 3. Deposit to balancer pool
+        uint256[] memory maxAmountsIn = new uint256[](2);
+        maxAmountsIn[0] = amount0;
+        maxAmountsIn[1] = amount1;
+
+        _addLiquidityBalancer(pool, tokens, maxAmountsIn, user, minOut);
+
+        emit PoolCreated(pool);
+    }
+
+    /**
+     * @dev Migrates a liquidity position, adds liquidity to the pool, and stake the BPT
+     *
+     * @param fromLpToken The LP token to migrate
+     * @param user The owner of the lp position
+     * @param tokens The underlying tokens of the pool.
+     * @param minOut The min amount of bpt
+     * @param pool The pool address to add liquidity
+     * @param rewardPoolAddress The aura reward pool address
+     */
+    function migrateUniswapV2AndJoinPool(
+        address fromLpToken,
+        address user,
```

Comment:
Why are we passing `user` here and not using `msg.sender`? Seems like there a risk here. Anyone who has given approval to this contract could be called here

Reply:
removed